shell/platform/android/platform_view_android.cc

```diff
@@ -104,6 +104,22 @@ void PlatformViewAndroid::NotifyCreated(
   PlatformView::NotifyCreated();
 }
 
+void PlatformViewAndroid::NotifySurfaceWindowChanged(
+    fml::RefPtr<AndroidNativeWindow> native_window) {
+  if (android_surface_) {
+    fml::AutoResetWaitableEvent latch;
+    fml::TaskRunner::RunNowOrPostTask(
+        task_runners_.GetRasterTaskRunner(),
+        [&latch, surface = android_surface_.get(),
+         native_window = std::move(native_window)]() {
+          surface->TeardownOnScreenContext();
+          surface->SetNativeWindow(native_window);
```

Comment:
this new native API just tears down the on screen context and sets the new native window.

Reply:
What is the difference between this and calling `FlutterRenderer.stopRenderingToSurface`/`FlutterRenderer.startRenderingToSurface`?

Reply:
`FlutterRenderer.stopRenderingToSurface()` calls `PlatformView::NotifyDestroyed()`, and then `Shell::OnPlatformViewDestroyed()`, which tears down the rasterizer: https://github.com/flutter/engine/blob/35d7d4aa6cf9560bf691f383c0a65cf982b65004/shell/common/shell.cc#L723 

It also causes [this issue](https://github.com/flutter/engine/blob/master/shell/gpu/gpu_surface_gl.cc#L215), which is due to [this method](https://github.com/google/skia/blob/c19b9c58862feee3a482575ae014293da02bda43/src/gpu/GrProxyProvider.cpp#L889) returning `true`. I asked about it on the Flutter & Skia chat, and @chinmaygarde had some context about previous cases.

 I haven't figured out exactly why the sequence `FlutterRenderer.stopRenderingToSurface`/`FlutterRenderer.startRenderingToSurface` is very different than just tearing down the `OnScreenContext` and then setting the native window.